code-dot-org/code-dot-org
# [ProgressV2] create a new feedback banner for progress table v2

## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.



https://github.com/code-dot-org/code-dot-org/assets/25193259/26f4c5e8-2829-4fef-b58e-90b2960c02cd


Shows a feedback banner for the new progress view.

Banner is only showed after the second time a user has clicked on the new progress view and only on the new progress view. Banner should only be shown until it has been answered for each user.

Based off of [this](https://github.com/code-dot-org/code-dot-org/pull/57517/files) PR, however we are not using local storage due to race conditions with loading feedback. This could be added later as an optimization.


Testing:
Unit tests will be added as a fast follow. We are trying to get this banner in before a release later today. Extensive manual testing was taken instead of unit tests and unit tests will be added ASAP.




## Links





## Testing story
Manual testing steps:
* banner is never shown on v1 table
* banner is not shown when `switch to new progress view` is clicked
* Banner is shown if v2 table page is hard reloaded
* Banner is not shown if feedback has already been given
* Answered banner shows the `share more` link
* Closed banner is never re-shown





## Deployment strategy

## Follow-up work



## Privacy



## Security



## Caching

## PR Checklist:
